### PR TITLE
[feat]:SignIn and SignUp functionality

### PR DESCRIPTION
**Pull request by ultronjs**
Integrated SignIn and SIngUp API and all the functionality related to the same.
Adding to Wishlist and Cart only if user is logged in.
Toast on Error and Info also Added

### CODE REVIEW

File: src/components/LoginForm.jsx [hunk 1]
```diff
@@ -1,8 +1,41 @@
...
+  const initialSignInState = {
+    email: "",
+    password: ""
+  };
+  const [signIn, setSignIn] = useState(initialSignInState);
+  const { signInStatusDispatch } = useAuth();
+  const onChangeHandler = (e) => {
+    setSignIn((prevState) => ({
+      ...prevState,
+      [e.target.name]: e.target.value,
```
Comment (JantuDeb) [inline src/components/LoginForm.jsx:24]: Liked that yo are using this dynamic  key

File: src/components/Toast.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        <i class={`far ${variantOfToast[type][1]} fa-2x toast_icon`}></i>
+        <div class="toast_content">
+          <span class="toast_heading">{type}</span>
+          <span class="toast_body">{msg}</span>
+        </div>
+        <i class="far fa-times-circle fa-lg toast_dismiss"></i>
+      </div>
+    </div>
+  );
+}
```
Comment (JantuDeb) [inline src/components/Toast.jsx:22]: Liked the way you are using toast

File: src/utils/axios.js [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
-  headers: {'Accept': '*/*',authorization: encodedToken}
+  headers: { Accept: "*/*", authorization: localStorage.getItem("token") },
+});
+privateInstance.interceptors.request.use(async (req) => {
+  console.log("intercerper", req);
+  return req;
+});
+privateInstance.interceptors.response.use(async (res) => {
+  console.log("intercerper response", res);
+  return res;
```
Comment (JantuDeb) [inline src/utils/axios.js]: Remove interceptors if you are not doing anything.

**Review by JantuDeb — APPROVED** [review]

File: src/utils/axios.js (hunk 3 above)
Comment (ultronjs) [reply src/utils/axios.js]: Done

File: src/context/wishListContext.js [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
 const WishListProvider = ({children}) => {
+    const {addToast} = useToast()
 
     const getWishListData = async () => {
-    console.log("i m getting triggered getWishList")
     try{
         const {status,data} = await privateInstance
       .get("/user/wishlist")
       if(status ===200){
         wishListDispatch({ type: "SET_DATA", payload: data.wishlist })
```
Comment (Priya-Patel31) [inline src/context/wishListContext.js:17]: Suggestion : here you can use payload: {wishlist : data.wishlist} that would be more understandable that which kind of data it is .

File: src/index.js [hunk 5]
```diff
@@ -4,22 +4,26 @@ import "./index.css";
...
-    <FilterProvider>
-      <CartProvider>
-        <WishListProvider>
-          <BrowserRouter>
-            <App />
-          </BrowserRouter>
-        </WishListProvider>
-      </CartProvider>
-    </FilterProvider>
+    <ToastProvider>
```
Comment (Priya-Patel31) [inline src/index.js:14]: You can have a separate file for these contexts and then add import that single component here .

File: src/reducers/filterReducer.js [hunk 6]
```diff
@@ -25,7 +24,6 @@ export const filterReducer = (state, action) => {
     case "sortByAccessoriesType":
       return { ...state, sortByAccessoriesType: action.payload };
     case "sortByBrands":
```
Comment (Priya-Patel31) [inline src/reducers/filterReducer.js:26]: Use uppercase for the name of these cases .

File: src/components/Toast.jsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+
+function Toast({type,msg}) {
```
Comment (Priya-Patel31) [inline src/components/Toast.jsx:3]: Really like the way you have made the dynamic toasts.

File: src/components/SignUpForm.jsx [hunk 8]
```diff
@@ -1,59 +1,116 @@
...
-          />
-          <i className="input_icon fas fa-eye-slash"></i>
-        </div>
-        <div className>
-          <button className="btn btn_primary">CREATE ACCOUNT</button>
+    <>
+      <div className="form_container">
+        <div className="form sign_up">
+          <div className="form_img">
+            <img src="/assets/login_logo.png" alt="" />
```
Comment (Priya-Patel31) [inline src/components/SignUpForm.jsx]: give some text in the alt tag.

**Review by Priya-Patel31 — APPROVED** [review]
The code is well formatted and really liked the toast which you have added . Also , liked the emptyCart and emptyWishlist components which you have added in your code . Just have few suggestions and minor changes you can go through them.

File: src/context/wishListContext.js (hunk 4 above)
Comment (ultronjs) [reply src/context/wishListContext.js:17]: Done

File: src/components/SignUpForm.jsx (hunk 8 above)
Comment (ultronjs) [reply src/components/SignUpForm.jsx]: Done

**Pull request merged**